packages/hasher/src/lockfile.ts

```diff
@@ -0,0 +1,91 @@
+import path from "path";
+import findUp from "find-up";
+import fs from "fs-extra";
+import { parse as parseYarnLock } from "@yarnpkg/lockfile";
+import { readWantedLockfile } from "@pnpm/lockfile-file";
+import { Dependencies } from "./resolveExternalDependencies";
+import { nameAtVersion } from "./helpers";
+
+type LockDependency = {
+  version: string;
+  dependencies?: Dependencies;
+};
+
+export type ParsedLock = {
+  type: "success" | "merge" | "conflict";
+  object: {
+    [key in string]: LockDependency;
+  };
+};
+
+const memoization: { [path: string]: ParsedLock } = {};
+
+export async function parseLockFile(packageRoot: string): Promise<ParsedLock> {
+  const yarnLockPath = await findUp("yarn.lock", { cwd: packageRoot });
+
+  // First, test out whether this works for yarn
```

Comment:
I feel a bit ambivalent about this. On one hand it's convenient not have to do this configuration, on the other, it seems good to explicit tell the tool what it is dealing with. I'm thinking of leaving that as is if it's okay with you since it will not have performance hit for you and will have negligible difference for pnpm / rush users